src/main/java/uk/gov/hmcts/divorce/common/service/task/GenerateAosResponseLetterDocument.java

```diff
@@ -54,9 +54,9 @@ public CaseDetails<CaseData, State> apply(CaseDetails<CaseData, State> caseDetai
         final AcknowledgementOfService acknowledgementOfService = caseData.getAcknowledgementOfService();
 
         if (caseData.getApplicant1().isApplicantOffline()) {
-            if (YES.equals(caseData.getIsJudicialSeparation())) {
+            if (caseData.isJudicialSeparationCase()) {
+                generateD84Form.generateD84Document(caseData, caseId);
```

Comment:
Moved this up one line, to ensure that it is always generated for JS cases regardless of whether they are disputed.